documentation/user_docs/docs/manual/Parallel.rst

```diff
@@ -0,0 +1,130 @@
+
+
+Controlling MPI
+===============
+
+Certain operations in Horace are parallelised with an MPI algorithm. These can be enabled in one go using:
+
+::
+
+   hpc('on')
+
+In order to manually enable/disable different parallel components it is possible, these are managed through the ``hpc_config``.
+
+::
+
+     hpc_config with properties:
+
+         build_sqw_in_parallel: 0
+       parallel_workers_number: 2
+             combine_sqw_using: 'matlab'
+           combine_sqw_options: {'matlab'  'mex_code'  'mpi_code'}
+       mex_combine_thread_mode: 0
+       mex_combine_buffer_size: 131072
+             parallel_multifit: 0
+              parallel_cluster: 'herbert'
+        parallel_configuration: [1x1 parallel_config]
+                   hpc_options: {1x6 cell}
+                    class_name: 'hpc_config'
+                      saveable: 1
+              returns_defaults: 0
+                 config_folder: '/home/jacob/.matlab/mprogs_config'
+
+In particular, the parallel enabling options are:
+- ``parallel_multifit`` : Enable parallel fitting for ``multifit`` and ``tobyfit``
+- ``build_sqw_in_parallel`` : Enable building sqw objects in parallel, i.e. ``gen_sqw``, ``combine_sqw``
+
+The ``parallel_config`` contains most of the information to manage parallelism:
+
+::
+
+      parallel_config with properties:
+
+                         worker: 'worker_v2'
+                    is_compiled: 0
+               parallel_cluster: 'herbert'
+                 cluster_config: 'local'
+                 known_clusters: {'herbert'  'parpool'  'mpiexec_mpi'  'slurm_mpi'  'dummy'}
+            known_clust_configs: {'local'}
+         shared_folder_on_local: ''
+        shared_folder_on_remote: ''
+              working_directory: '/tmp/'
+               wkdir_is_default: 1
+               external_mpiexec: ''
+                     class_name: 'parallel_config'
+                       saveable: 1
+               returns_defaults: 0
+                  config_folder: '/home/jacob/.matlab/mprogs_config'
+
+MPI Schemes
+===========
+
+Horace can be run in parallel with a number of different schemes, all controlled through the ``parallel_config``.
+
+The five currently implemented parallel schemes are:
+1. ``herbert`` (Poor-man's MPI) - Data messages are sent through files written to the hard drive and read by each
+process. This is the slowest MPI scheme, but also the one with the fewest requirements.
+2. ``parpool`` (Matlab Parallel Toolbox MPI) - Parpool uses Matlab's parallel toolbox parallelism to send messages and
+therefore requires the parallel toolbox to be used.
+3. ``mpiexec_mpi`` (C++ MPI) - Data messages are sent using C++ wrapping OpenMPI. This requires the MEX files to be
+built in order to be used.
+4. ``slurm_mpi`` (Slurm MPI) - Data messages are sent using C++ wrapping OpenMPI, but are submitted to a running slurm
+instance by Horace upon starting the job. This requires the MEX files to be built in order to be used.
```

Comment:
Does this mean that parallelism only works for mex-ported functions, or is the mex just for the comms?